databricks/koalas/frame.py

```diff
@@ -10220,12 +10220,22 @@ def _to_internal_pandas(self):
         """
         return self._internal.to_pandas_frame
 
+    def _get_or_create_repr_pandas_cache(self, n):
+        if (
+            not hasattr(self, "_repr_pandas_cache")
+            or (id(self._internal), n) not in self._repr_pandas_cache
+        ):
+            self._repr_pandas_cache = {
+                (id(self._internal), n): self.head(n + 1)._to_internal_pandas()
+            }
+        return self._repr_pandas_cache[(id(self._internal), n)]
```

Comment:
Hm ... @ueshin should we maybe just keep one cache just to prevent the possibility of the growing cache?

Reply:
It actually keeps one cache by the key `(id(self._internal), n)`.

Reply:
I meant, it keeps the cache if we change `display.max_rows` or operate that change `_internal`.
For example, if somebody opens a shell, and does a lot of `inplace` operations with printing out in the notebook, the cache will keep growing. I was wondering if it's better to don't keep previous caches.

Reply:
I feel like at least we should have a logic to drop the old caches conditionally if we want to keep old ones.

Reply:
It does hold only one cache.
If `inplace` operation is done, `id(self._internal)` will change, and if `display.max_rows` is changed, `n` will change.